File: 645. Set Mismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function(nums) {
    let a = new Set();
    let b = 0;
    let output = [];
    
    for (let i = 0; i < nums.length; i++) {
        if (a.has(nums[i])) {
            b = nums[i];
            output[0] = b;
        } else {
            a.add(nums[i]);
        }
    }
    
    for (let k = 1; i < nums.length + 1; k++) {
         if (!a.has(k)) {
             output[1] = k;
             return output;
         }
    }
    
    return output;
};

File: 645. Set Mismatch.py
from collections import Counter
class Solution:
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        counted = Counter(nums)
        output = [num for num, count in counted.items() if count == 2]
        
        for i in range(1, len(nums) + 1):
            if i not in counted:
                output.append(i)
                return output
